tests/system/test_client.py

```diff
@@ -859,6 +859,67 @@ def test_load_table_from_json_basic_use(self):
         self.assertEqual(tuple(table.schema), table_schema)
         self.assertEqual(table.num_rows, 2)
 
+    def test_load_table_from_json_bug_check_with_schema(self):
+        json_rows = [
+            {
+                "age": "18",
+            },
+            {
+                "age": "79",
+            },
+        ]
+
+        dataset_id = _make_dataset_id("bq_system_test")
+        self.temp_dataset(dataset_id)
+        table_id = "{}.{}.load_table_from_json_bug_check_with_schema".format(
+            Config.CLIENT.project, dataset_id
+        )
+
+        # create an empty table with schema
+        table = helpers.retry_403(Config.CLIENT.create_table)(Table(table_id))
+        self.to_delete.insert(0, table)
+
+        job_config = bigquery.LoadJobConfig()
+        load_job = Config.CLIENT.load_table_from_json(
+            json_rows, table_id, job_config=job_config
+        )
+        load_job.result()
+
+        table = Config.CLIENT.get_table(table)
+        # schema should be inferred from the json data
+        self.assertEqual(
+            tuple(table.schema),
+            (bigquery.SchemaField("age", "STRING", mode="NULLABLE"),),
+        )
```

Comment:
I think we need to add an assert checking that the row values were correctly inserted as strings.